src/execute.rs

```diff
@@ -426,10 +426,346 @@ impl Execute for System {
+        self.write(address, result);
+
+        Ok(cycles + 2)
+      }
+
+      0xE3 | 0xE7 | 0xEF | 0xF3 | 0xF7 | 0xFB | 0xFF => {
+        // ISC: INC => SBC
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = self.read(address);
+        let result = value.wrapping_add(1);
+        self.registers.alu_subtract(value);
+        self.registers.sr.set_nz(result);
+        self.write(address, result);
+
+        Ok(cycles + 2)
+      }
+
+      0x0B | 0x2B => {
+        // ANC: AND byte with accumulator. If result is negative then carry is set.
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let new_val = self.registers.a & value;
+        self.registers.sr.write(flags::CARRY, new_val & 0x80 != 0);
+
+        Ok(cycles)
+      }
+
+      0x4B => {
+        // ALR: AND + LSR
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let new_val = (self.registers.a & value) >> 1;
+
+        self.registers.sr.write(flags::CARRY, new_val & 0x01 != 0);
+        self.registers.sr.set_nz(new_val);
+
+        Ok(cycles)
+      }
+
+      0x6B => {
+        // ARR: AND + ROR
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let mut new_val = self.registers.a & value;
+
+        let new_val = (new_val >> 1) | (self.registers.sr.read(flags::CARRY) as u8) << 7;
+
+        self.registers.sr.write(flags::CARRY, new_val & 0x40 != 0);
+        self
+          .registers
+          .sr
+          .write(flags::OVERFLOW, new_val & 0x20 != 0);
+        self.registers.sr.set_nz(new_val);
+
+        Ok(cycles)
+      }
+
+      0x8B => {
+        // XAA: AND X + AND immediate
+        // Oooo she's highly unstable xx "Do not use" or whatever
+
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        let magic: u8;
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+          magic = rand::random::<u8>();
+        }
+        #[cfg(target_arch = "wasm32")]
+        {
+          magic = 0xFF;
+        }
+        self.registers.a |= magic;
+        self.registers.a &= self.registers.x & value;
+        self.registers.sr.set_nz(self.registers.a);
+
+        Ok(cycles)
+      }
+
+      0xCB => {
+        // AXS: AND -> DEX -> STX
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        self.registers.x &= self.registers.a;
+
+        self.registers.sr.set(flags::CARRY);
+        self.registers.x = self.registers.x.wrapping_sub(value);
+        self.registers.sr.set_nz(self.registers.x);
+
+        Ok(cycles)
+      }
+
+      0xEB => {
+        // SBC (same as official sbc)
+        let (value, cycles) = self.fetch_operand_value(opcode);
+        self.registers.alu_subtract(value);
+        Ok(cycles)
+      }
+
+      0x9C => {
+        // SHY: (Y & (high(addr) + 1)) -> addr
+        let (address, cycles) = self.fetch_operand_address(opcode);
+        let value = address.to_be_bytes().iter().next().unwrap().clone();
```

Comment:
```suggestion
        let value = (address >> 8) as u8;
```

Iterators are relatively slow, and there's a much faster way to get the high byte of a 16-bit value

(This is relevant for the following instructions as well)